docs/manual/developer/03_creating_content.md

```diff
@@ -355,3 +355,372 @@ A profile should define these attributes:
+    notes: >-
+      Certain period of inactivity is vague.
+  - id: R2
+    title: Minimization of configuration
+    description: >-
+      The features configured at the level of launched services
+      should be limited to the strict minimum.
+    automated: no
+    note: >-
+      This is individual depending on the system workload
+      therefore needs to be audited manually.
+    related_rules:
+       - systemd_target_multi_user
+  - id: R3
+    title: Enabling SELinux targeted Policy
+    description: >-
+      It is recommended to enable SELinux in enforcing mode
+      and to use the targeted policy.
+    automated: yes
+    rules:
+      - selinux_state
+  - id: R4
+    title: Configure authentication
+    description: >-
+      Ensure authentication methods are functional to prevent
+      unauthorized access to the system.
+    controls:
+      - id: R4.a
+        title: Disable administrator accounts
+        automated: yes
+        level: low
+        rules:
+          -  accounts_passwords_pam_faillock_deny_root
+      - id: R4.b
+        title: Enforce password quality standards
+        automated: yes
+        level: high
+        rules:
+          - accounts_password_pam_minlen
+          - accounts_password_pam_ocredit
+          - var_password_pam_ocredit=1
+```
+
+### Using controls in profiles
+
+Later, we can use the policy requirements in profile YAML. Let’s say that we
+will define a “Desktop” profile built from the controls.
+
+To use controls, we add them under `selection` keys in the profile. The entry
+has the form `policy_id:control_id`.
+
+```
+$ cat rhel8/profiles/abcd-desktop.profile
+ 
+documentation_complete: true
+title: ABCD Desktop for Red Hat Enterprise Linux 8
+description: |-
+  This profile contains configuration checks that align to
+  the ABCD benchmark.
+selections:
+  - abcd:R1
+  - abcd:R2
+  - abcd:R3
+  - security_patches_uptodate
+```
+
+Notice we can mix the controls selections with normal rule selections.
+
+In a similar way, we could define another profile that selects only some of the
+requirements.
+
+In the example we have selected all controls from `controls/abcd.yml` by listing
+them explicitly. It is possible to shorten it using the `“all”` value which
+means that all controls will be selected. Let’s show how it will be easier:
+
+```
+$ cat rhel8/profiles/abcd-high.profile
+ 
+documentation_complete: true
+title: ABCD High for Red Hat Enterprise Linux 8
+description: |-
+  This profile contains configuration checks that align to
+  the ABCD benchmark.
+selections:
+  - abcd:all
+  - security_patches_uptodate
+```
+
+It is possible to use levels if the levels are defined in the controls file. For
+example, `abcd:all:low` selects all rules for the ABCD low level or
+`abcd:all:high` selects all rules from the ABCD high level.
+
+Finally, when we build the content we will automatically get a SCAP profile
+which contains XCCDF rules and variables from all controls selected in profile
```

Comment:
This isn't explicitly mentioned above so I figured I'd call it out: `variables` is only discussed here. Can you select variables in the control file's `rules` section? Should that section be renamed if it isn't strictly only rules and is more or less similar to the Profile shorthand? What if they differ based on level?

e.g., Level 1 Server/Workstation in CIS would take AppArmor EnforceOrComplain variable with either value (enforce or complain) -- but Level 2 Server/Workstation could only take Enforce setting.

Would we just ignore the value of the variable in level 2? &c.

Reply:
Yes, variables can be selected. In one of the examples we select `var_accounts_tmout=10_min`. It's more or less the same as the profile selections. I agree the key should be renamed, ideally to `selections`. It isn't possible to have the different selections in different levels at this moment. It would make sense to add the option to have it somehow.